pkg/reconciler/ingress/ingress.go

```diff
@@ -578,3 +583,19 @@ func getLBStatus(gatewayServiceURL string) []v1alpha1.LoadBalancerIngressStatus
 func enablesAutoTLS(ctx context.Context) bool {
 	return config.FromContext(ctx).Network.AutoTLS
 }
+func isIstioInjectionEnabledForNs(r *BaseIngressReconciler, ra ReconcilerAccessor, ia v1alpha1.IngressAccessor) (bool, error) {
+	ing, err := ra.GetIngress(ia.GetNamespace(), ia.GetName())
+	if err != nil {
+		return false, err
+	}
+	ns := ing.GetLabels()[serving.RouteNamespaceLabelKey]
+	realNs, err := r.KubeClientSet.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	inject := realNs.Labels["istio-injection"]
```

Comment:
probably should be made constant.
Also all five lines can be collapsed into single `realNs.Labels["istio-injection"] =="enabled", nil`.